packages/opentelemetry-tracing/src/BasicTracerProvider.ts

```diff
@@ -32,10 +36,14 @@ export class BasicTracerProvider implements api.TracerProvider {
 
   activeSpanProcessor = new NoopSpanProcessor();
   readonly logger: api.Logger;
+  readonly sampler: api.Sampler;
   readonly resource: Resource;
 
   constructor(private _config: TracerConfig = DEFAULT_CONFIG) {
     this.logger = _config.logger || new ConsoleLogger(_config.logLevel);
+    this.sampler =
+      _config.sampler ||
```

Comment:
by default `_config` is the `DEFAULT_CONFIG` so your fallback is never called here

Reply:
Thanks @vmarchaud I did this because of ts error:

```
Type 'Sampler | undefined' is not assignable to type 'Sampler'.
  Type 'undefined' is not assignable to type 'Sampler'.
```

But have worked around this with:

`  readonly sampler: api.Sampler | undefined;`

Reply:
We should always have a sampler though so you approach was fine, however i would use `??` instead of `||`